app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -229,8 +229,8 @@ def instances
       status = instance.fetch_path('state', 'name')
       next if collector.options.ignore_terminated_instances && status.to_sym == :terminated
 
-      # TODO(lsmola) we have a non lazy dependency, can we remove that?
-      flavor = persister.flavors.find(instance['instance_type']) || persister.flavors.find("unknown")
```

Comment:
`persister.flavors.find("unknown")` <- this is the issue if the `"unknown"` flavor gets deleted it returns `nil` and doesn't rebuild it as part of the targeted refresh.

If we include the flavors in the targeted refresh then it doesn't matter what gets deleted they'll be re-created properly